src/sdk/namespace_fs.js

```diff
@@ -3223,33 +3262,30 @@ class NamespaceFS {
     async _delete_latest_version(fs_context, latest_ver_path, params) {
         dbg.log0('Namespace_fs._delete_latest_version:', latest_ver_path, params);
 
-        let gpfs_options;
+        let files;
         const is_dir_content = this._is_directory_content(latest_ver_path, params.key);
-        const is_gpfs = native_fs_utils._is_gpfs(fs_context);
         let retries = config.NSFS_RENAME_RETRIES;
         let latest_ver_info;
         let versioned_path;
         for (;;) {
             try {
-                // TODO get latest version from file in POSIX like in GPFS path
                 latest_ver_info = await this._get_version_info(fs_context, latest_ver_path);
                 dbg.log1('Namespace_fs._delete_latest_version:', latest_ver_info);
                 if (latest_ver_info) {
-                    if (is_gpfs) {
-                    gpfs_options = await this._open_files_gpfs(fs_context, latest_ver_path, undefined, undefined, undefined,
-                            undefined, true);
-                        const latest_fd = gpfs_options?.delete_version?.src_file;
-                        latest_ver_info = latest_fd && await this._get_version_info(fs_context, undefined, latest_fd);
-                        if (!latest_ver_info) break;
-                    }
+                    files = await this._open_files(fs_context, { src_path: latest_ver_path, delete_version: true });
```

Comment:
again, should probably not open on non GPFS non lifecycle paths

Reply:
I agree, I did a lot of refactoring in this PR and I think we don't want to take a risk at this point of time
moving it to a different PR